### PR TITLE
fix(api): Remove nonexistent metric alert owners when transferring projects

### PR DESCRIPTION
**Pull request by snigdhas**
Remove nonexistent owners for metric alerts when transferring a projects

Transferring projects with alert owners can result in orphaned alerts if the owners are not a part of the new project. We should remove the alert owners before proceeding with the transfer. 

Fixes WOR-1855

### CODE REVIEW

File: src/sentry/api/endpoints/project_transfer.py [hunk 1]
```diff
@@ -63,6 +64,17 @@ def post(self, request: Request, project) -> Response:
...
+        old_org = Organization.objects.get(id=project.organization.id)
+        alerts = AlertRule.objects.filter(organization=old_org)
+        for alert in alerts:
+            member = OrganizationMember.objects.filter(
+                organization=owner.organization, id=alert.owner_id
+            ).first()
+
+            # Remove the alert owner if the owner is not in the new organization
+            if member is None:
+                alert.update(owner=None)
```
Comment (wedamija) [inline src/sentry/api/endpoints/project_transfer.py:76]: This is where we make the request to transfer a project. The actual transfer happens once someone views the email and clicks the approve button, which eventually calls https://github.com/getsentry/sentry/blob/4041b4258e62178c4314980811245391be4e28d5/src/sentry/models/project.py#L265

File: src/sentry/api/endpoints/project_transfer.py [hunk 2]
```diff
@@ -63,6 +64,17 @@ def post(self, request: Request, project) -> Response:
                 status=status.HTTP_404_NOT_FOUND,
             )
 
+        old_org = Organization.objects.get(id=project.organization.id)
+        alerts = AlertRule.objects.filter(organization=old_org)
+        for alert in alerts:
+            member = OrganizationMember.objects.filter(
+                organization=owner.organization, id=alert.owner_id
+            ).first()
```
Comment (wedamija) [inline src/sentry/api/endpoints/project_transfer.py:72]: `AlertRule.owner` is an `Actor`, not a `User`. `Actor` is basically a stand in to say "either `User` or `Team`".

So you would want to do something roughly like
```
if alert.owner:
    owner = alert.owner.resolve # Gives you either a User or a Team
    valid_owner = False
    if isinstance(owner, User) and OrganizationMember.objects.filter(organization=owner.organization, id=owner.id).exists():
        valid_owner = True
    # From a quick glance we just don't transfer over teams, so any alert assigned to a team should just have the owner removed.
    if not valid_owner:
        alert.update(owner=None)
        
    
```

**Review by wedamija — COMMENTED** [review]
We probably want to do a backfill here for the small number of rows that have invalid owners: https://redash.getsentry.net/queries/2740/source

@barkbarkimashark did something similar recently, could you dig up the pr?

File: src/sentry/api/endpoints/project_transfer.py (hunk 1 above)
Comment (snigdhas) [reply src/sentry/api/endpoints/project_transfer.py:76]: Gotcha. Removing the alert owners should probably happen once the transfer is approved.

Comment (barkbarkimashark) [reply src/sentry/api/endpoints/project_transfer.py:76]: This is the boy?
https://github.com/getsentry/sentry/pull/33488

Comment (snigdhas) [reply src/sentry/api/endpoints/project_transfer.py:76]: Looks correct! I think this PR is redundant with #29979, so I'll close the PR and handle the backfill to resolve the ticket. Thanks!

**Pull request closed without merging**